### PR TITLE
Proj0020 Item group should only contain nodes of a single type

### PR DESCRIPTION
**Pull request by CptWesley**
Documentation PR: https://github.com/dotnet-project-file-analyzers/dotnet-project-file-analyzers.github.io/pull/3

### CODE REVIEW

File: projects/MixedItemGroup/Palceholder.cs [hunk 1]
```diff
@@ -0,0 +1,4 @@
+﻿namespace MixedItemGroup;
```
Comment (LauraXiulan) [inline projects/MixedItemGroup/Palceholder.cs:1]: Typo in filename

**Review by Corniel — CHANGES_REQUESTED** [review]
See remarks in https://github.com/dotnet-project-file-analyzers/dotnet-project-file-analyzers.github.io/pull/3: This rule needs some extra thoughts.

File: projects/MixedItemGroup/Palceholder.cs (hunk 1 above)
Comment (CptWesley) [reply projects/MixedItemGroup/Palceholder.cs]: Fixed

**Comment by CptWesley** [comment]
> See remarks in [dotnet-project-file-analyzers/dotnet-project-file-analyzers.github.io#3](https://github.com/dotnet-project-file-analyzers/dotnet-project-file-analyzers.github.io/pull/3): This rule needs some extra thoughts.

Updated implementation

File: src/DotNetProjectFile.Analyzers/Analyzers/MsBuild/ItemGroupShouldBeUniform.cs [hunk 2]
```diff
@@ -0,0 +1,75 @@
+﻿namespace DotNetProjectFile.Analyzers.MsBuild;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+public sealed class ItemGroupShouldBeUniform : MsBuildProjectFileAnalyzer
+{
+    private static readonly IReadOnlyCollection<IReadOnlyCollection<string>> Exceptions =
+        [
+            [
+                "Compile",
```
Comment (Corniel) [inline src/DotNetProjectFile.Analyzers/Analyzers/MsBuild/ItemGroupShouldBeUniform.cs:9]: Might be worth to define these as nodes, and do on the base of types instead.

**Review by Corniel — APPROVED** [review]
Besides a small remark, LGTM.

File: src/DotNetProjectFile.Analyzers/Analyzers/MsBuild/ItemGroupShouldBeUniform.cs (hunk 2 above)
Comment (CptWesley) [reply src/DotNetProjectFile.Analyzers/Analyzers/MsBuild/ItemGroupShouldBeUniform.cs]: Done

**Review by Corniel — APPROVED** [review]

**Pull request merged**